File: src/main/java/com/willmadison/legacycodekatas/fulfillment/warehouse/exceptions/configuration/ExceptionConfiguration.java
package com.willmadison.legacycodekatas.fulfillment.warehouse.exceptions.configuration;

import com.willmadison.legacycodekatas.fulfillment.orders.Order;

import java.util.EnumSet;
import java.util.Set;

public class ExceptionConfiguration {

    public boolean enabled;

    public boolean warehouseOperational;

    public Set<Order.Type> supportedOrderTypes = EnumSet.of(Order.Type.B2B, Order.Type.B2C, Order.Type.LARGE_BULKY_ITEM);

    public int maxAutoStraggles = 5;

    public long autoStraggleTimeframeMinutes = 45L;

    public boolean autoStraggleEnabled = false;

    public boolean isEnabled() {
        return enabled;
    }

    public void setEnabled(boolean enabled) {
        this.enabled = enabled;
    }

    public boolean isWarehouseOperational() {
        return warehouseOperational;
    }

    public void setWarehouseOperational(boolean warehouseOperational) {
        this.warehouseOperational = warehouseOperational;
    }
}

File: src/main/kotlin/com/willmadison/legacycodekatas/fulfillment/warehouse/exceptions/configuration/ExceptionConfiguration.kt
package com.willmadison.legacycodekatas.fulfillment.warehouse.exceptions.configuration

import com.willmadison.legacycodekatas.fulfillment.orders.OrderType

open class ExceptionConfiguration(var enabled: Boolean = true,
                                  var warehouseOperational: Boolean = true,
                                  var supportedOrderTypes: Set<OrderType> = OrderType.values().toSet(),
                                  var maxAutoStraggles: Int = 5,
                                  val autoStraggleTimeframeMinutes: Long = 45,
                                  val autoStraggleEnabled: Boolean = false
)